rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -311,6 +408,198 @@ def _tx_swaps_to_trades(swaps: List[AMMSwap]) -> List[AMMTrade]:
 
         return trades
 
+    def _get_events_balances(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+    ) -> AddressEventsBalances:
+        """Request via graph all events for new addresses and the latest ones
+        for already existing addresses. Then the requested events are written
+        in DB and finally all DB events are read, and processed for calculating
+        total profit/loss per LP (stored within <LiquidityPoolEventsBalance>).
+        """
+        address_events_balances: AddressEventsBalances = {}
+        address_events: DDAddressEvents = defaultdict(list)
+        db_address_events: AddressEvents = {}
+        new_addresses: List[ChecksumEthAddress] = []
+        existing_addresses: List[ChecksumEthAddress] = []
+        min_end_ts: Timestamp = to_timestamp
+
+        # Get addresses' last used query range for Uniswap events
+        for address in addresses:
+            entry_name = f'{UNISWAP_EVENTS_PREFIX}_{address}'
+            events_range = self.database.get_used_query_range(name=entry_name)
+
+            if not events_range:
+                new_addresses.append(address)
+            else:
+                existing_addresses.append(address)
+                min_end_ts = min(min_end_ts, events_range[1])
+
+        # Request new addresses' events
+        if new_addresses:
+            start_ts = Timestamp(0)
+            for address in new_addresses:
+                for event_type in EventType:
+                    new_address_events = self._get_events_graph(
+                        address=address,
+                        start_ts=start_ts,
+                        end_ts=to_timestamp,
+                        event_type=event_type,
+                    )
+                    if new_address_events:
+                        address_events[address].extend(new_address_events)
+
+                # Insert new address' last used query range
+                self.database.update_used_query_range(
+                    name=f'{UNISWAP_EVENTS_PREFIX}_{address}',
+                    start_ts=start_ts,
+                    end_ts=to_timestamp,
+                )
+
+        # Request existing DB addresses' events
+        if existing_addresses and min_end_ts <= to_timestamp:
+            for address in existing_addresses:
+                for event_type in EventType:
+                    address_new_events = self._get_events_graph(
+                        address=address,
+                        start_ts=min_end_ts,
+                        end_ts=to_timestamp,
+                        event_type=event_type,
+                    )
+                    if address_new_events:
+                        address_events[address].extend(address_new_events)
+
+                # Update existing address' last used query range
+                self.database.update_used_query_range(
```

Comment:
some similarities between this logic, and the trades one. And I guess we will see common similarities with other protocols.

We should try to abstract this common functionality.

This is just a note. Nothing required to do in this PR.

Reply:
@vnavascues can you make an issue about this so that we don't forget it?